src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -691,11 +691,6 @@ internal void EnsureStoreIsValid()
             if (Disposed)
                 throw new ObjectDisposedException(null, SR.IsolatedStorage_StoreNotOpen);
 
-            if (IsDeleted)
-            {
-                throw new IsolatedStorageException(SR.IsolatedStorage_StoreNotOpen);
-            }
-
             if (_closed)
```

Comment:
I thought it was always true as that is how Desktop has it (never checks for deleted) and the tests passed. Did we ported something differently than desktop? I will then update the PR to do the double check as @danmosemsft pointed out just to make sure we don't introduce a bug.